### PR TITLE
#1515 Parameter fixes

### LINKED ISSUE #1515: Mistakes in last shuffle round of code
### `SupplierInvoices`
- New supplier invoice button is broken, not passing the supplier invoices route
### `CustomerInvoicePage`
- Add from master list not working
- page info not working
### `CustomerRequisitionPage`
- Set supplied and set requested buttons not working
- page info not working
### `StocktakeManagePage`
- filtering not working
### `SupplierInvoicePage`
- page info not working

### Version and device info

- App version: 3.0.3 DEVELOPMENT
- Tablet model: All
- OS version: All

### Additional context

Simple one lines

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1515 

## Change summary

- Fixes parameter mistakes in each of the pages listed in the issue

## Testing

- [ ] Can add a new supplier invoice
- [ ] Can add from a master list in customer invoice page
- [ ] Can add a comment in customer invoice page, customer requisition page, supplier invoice page.
- [ ] can filter items in stocktake manage page

### Related areas to think about

N/A

### CODE REVIEW

**Comment by Chris-Petty** [comment]
- [x] Can add a new supplier invoice
- [x] Can add from a master list in customer invoice page
- [x] Can add a comment in customer invoice page, customer requisition page, supplier invoice page.
- [x] can filter items in stocktake manage page

> ### `StocktakeManagePage`
> * filtering not working
> 
I can't reproduce on branch `develop`, but at any rate it's "fixed"

__I cannot delete customer invoices, supplier invoices or supplier requisitions__

File: src/reducers/PagesReducer.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ export const PagesReducer = (state = {}, action) => {
       const { pageObject } = params;
 
       const pageInitialiser = getPageInitialiser(routeName);
-      const pageInitialState = pageInitialiser(pageObject);
+      const pageInitialState = pageInitialiser?.(pageObject);
```
Comment (Chris-Petty) [inline src/reducers/PagesReducer.js]: Is this valid?

File: src/pages/SupplierInvoicePage.js [hunk 2]
```diff
@@ -215,4 +216,5 @@ SupplierInvoice.propTypes = {
   onEditTotalQuantity: PropTypes.func.isRequired,
   onEditDate: PropTypes.func.isRequired,
   onAddTransactionBatch: PropTypes.func.isRequired,
+  route: PropTypes.func.isRequired,
```
Comment (Chris-Petty) [inline src/pages/SupplierInvoicePage.js]:
```suggestion
  route: PropTypes.string.isRequired,
```

File: src/reducers/PagesReducer.js (hunk 1 above)
Comment (joshxg) [reply src/reducers/PagesReducer.js]: yes: https://github.com/tc39/proposal-optional-chaining

others too i.e. https://github.com/tc39/proposal-nullish-coalescing

pr added: https://github.com/openmsupply/mobile/pull/1393

File: src/pages/SupplierInvoicePage.js (hunk 2 above)
Comment (joshxg) [reply src/pages/SupplierInvoicePage.js]: 7767ebf

**Comment by joshxg** [comment]
will make an issue for the deleting

File: src/reducers/PagesReducer.js (hunk 1 above)
Comment (Chris-Petty) [reply src/reducers/PagesReducer.js]: Nice, I like it!

Comment (Chris-Petty) [reply src/reducers/PagesReducer.js]: https://ponyfoo.com/articles/null-propagation-operator

**Pull request merged**